doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownMarkup.java

```diff
@@ -0,0 +1,144 @@
+package org.apache.maven.doxia.module.markdown;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.maven.doxia.markup.TextMarkup;
+import org.codehaus.plexus.util.StringUtils;
+
+/**
+ * This interface defines all markups and syntaxes used by the <b>Markdown</b> format.
+ */
+@SuppressWarnings( "checkstyle:interfaceistype" )
+public interface MarkdownMarkup
+    extends TextMarkup
+{
+    // ----------------------------------------------------------------------
+    // Markup separators
+    // ----------------------------------------------------------------------
+
+    /** APT backslash markup char: '\\' */
+    char BACKSLASH = '\\';
+
+    String COMMENT_START = "<!-- ";
+    String COMMENT_END = " -->";
+
+    /** APT numbering decimal markup char: '1' */
+    char NUMBERING = '1';
+
+    /** APT numbering lower alpha markup char: 'a' */
+    char NUMBERING_LOWER_ALPHA_CHAR = 'a';
+
+    /** APT numbering lower roman markup char: 'i' */
+    char NUMBERING_LOWER_ROMAN_CHAR = 'i';
+
+    /** APT numbering upper alpha markup char: 'A' */
+    char NUMBERING_UPPER_ALPHA_CHAR = 'A';
+
+    /** APT numbering upper roman markup char: 'I' */
+    char NUMBERING_UPPER_ROMAN_CHAR = 'I';
+
+    /** APT page break markup char: '\f' */
+    char PAGE_BREAK = '\f';
+
+    /** APT percent markup char: '%' */
+    char PERCENT = '%';
+
+    /** APT tab markup char: '\t' */
+    char TAB = '\t';
+
+    // ----------------------------------------------------------------------
+    // Markup syntax
+    // ----------------------------------------------------------------------
+
+    /** Syntax for the anchor end: "\"></a>" */
+    String ANCHOR_END_MARKUP = "\"></a>";
+
+    /** Syntax for the anchor start: "<a name=\"" */
+    String ANCHOR_START_MARKUP = "<a name=\"";
```

Comment:
Markdown will never use `name` for anchors. Only `id`.

Reply:
currently, code is in Doxia 1, which is still using name, isn't it?
you're right that id will be required when merging to Doxia 2

Reply:
Yes, correct. You need to keep an eye on the changes from 1 to 2.

Reply:
I'll let you merge from 1 to 2 :)
I don't think there are many changes

Reply:
Grrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrr....